src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/common.hpp

```diff
@@ -155,21 +204,30 @@ static constexpr size_t vec_len_f32_neon = vec_len_neon / sizeof(float);
         _mm_storeu_si128(reinterpret_cast<__m128i *>(addr), bf16_o);
     }
 
-    inline __m256 mm256_uni_loadu_ps(ov::float16* a) {
-        auto vec_f16 = _mm_loadu_si128(reinterpret_cast<__m128i*>(a));
-        auto o = _mm256_cvtph_ps(vec_f16);
-        return o;
-    }
-    inline __m256 mm256_uni_loadu_tail_ps(const ov::float16* a, const size_t count) {
-        ov::float16 tmp_values[8] = {0};
-        std::memcpy(tmp_values, a, count * sizeof(ov::float16));
-        return mm256_uni_loadu_ps(tmp_values);
-    }
     inline void mm256_uni_storeu_ps(ov::float16* a,  __m256 v) {
         __m128i vec_f16 = _mm256_cvtps_ph(v, 0);
         _mm_storeu_si128(reinterpret_cast<__m128i *>(a), vec_f16);
     }
 
+    // store __m256 to addr
+    inline void mm256_uni_storeu_tail_ps(float *addr, __m256 v, size_t count) {
+        const auto mask = get_mask(count);
+        return _mm256_maskstore_ps(addr, mask, v);
```

Comment:
This intrinsic is available in avx512 only. You cannot use it for avx2 code-path

Reply:
`_mm256_maskstore_ps` will be converted to `vmaskmovps` which should be exist in [AVX](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#ig_expand=5312,3117,5354,5638,4165,2329,4675,4675,2840,7545,7197,7208,7200,7197,4786,907,551,2329,1618,3335,4484,2232,2672,3434,3485,4646,504,713,1887,2689,6951,1878,1619,2520,2499,1947,2125,2073,4918,7084,4895,4918,1640,6596,506,4083,1641,699,2959,2977,2978,4989,4993,4994,5014,5018,6118,4588,536,598,2871,2961,1927,1926,2727,1641,1640,2701,2707,2707,4635,1762,4635,2945,4279,4280,4281,1690,1690,1690,3981,4857,4880,4871,4112,4088,4111,7030,7033,4111,1866,1664,7130,7131,6039,1866,3103,1511,1622,2174,4871,1640,2502,1869,4111,750,92,4030,1866,1664,3107,5749,750,1920,1929,6551,6551,4111,4044,140,607,668,822,3062,4360,4687,5821,6633,2245,1866,7131,2055,1916,1916,2055,2110,2052,4208,2738,4252,4263,4603,6220,460,4263,2227,6408,1753,1664,6875,1774,6176,1774,5822,545,1869,1667,5303,2955,2246,694,4019,6568,6555,5152,750,512,6877,6882,6883,5152,5152,5154,5155,6863,1931,1919,1572,6518,4082,6550,4975,5062,1572,2317,6550,5050,3878,691,7009,4108,5821,3111,6517,6053,6050,4928,5056,4928,6027,4930,7054,7054,7057,4986,7057,6015,4986,4934,4927,5053,5053,4953,4953,4954,4940,6015,6053,6033,2946,4954,4954,6047,4949,4954,4949,4954,2664,6015,4930,6015,4987,5062,5002,5062,6561,6560,6564,2570,2107,2110,1869,1667,1929,2668,2123,1918,1897,1920,1900,1923,1911,4106,1869,1667,574,5788,2632,2958,1927,1926,1900,1919,4030,4016,4108,4108,5643,6248,564,3841,4064,4110,4110,6233,5643,244,5665,5647,828,4263&amxtechs=AMX_BF16,AMX_INT8,AMX_TILE,AMX_FP16,AMX_COMPLEX&techs=SSE_ALL,AVX_ALL,AVX_512&text=_mm256_maskstore_ps).